### PR TITLE
Spellcheck outside git

### PR DESCRIPTION
**Pull request by lslezak**
- Allow running the `check:spelling` task outside of a Git clone
- This will allow to run it in Jenkins builds during package build (the tarball obviously does not contain a full Git clone)
- Do not use `git ls-files` for obtaining files to check

### CODE REVIEW

File: lib/tasks/spellcheck_task.rb [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+        end
+      end
+
+    end
+
+    private
+
+    # optionally colorize the misspelled words if the rainbow gem is present
+    # @return [Boolean] true when the colorization support is present
+    def colorize
```
Comment (jreidinger) [inline lib/tasks/spellcheck_task.rb]: better use `colorize?`

**Comment by jreidinger** [comment]
in general it looks good, just we need to change all projects that already use spellchecking

File: lib/tasks/spellcheck_task.rb (hunk 1 above)
Comment (lslezak) [inline lib/tasks/spellcheck_task.rb]: Well, I was thinking about this, it is actually a read accessor for `@colorize` attribute, the convention is to name it as the attribute. On the other hands it's a boolean flag so the question mark name would be OK...

I'm not sure which rule should we use in this case...  :grey_question: :grey_question: :grey_question:

**Comment by lslezak** [comment]
Yes, I'll adopt all modules which use the `check:spelling` task, there are just few of them...

File: lib/tasks/spellcheck_task.rb (hunk 1 above)
Comment (jreidinger) [inline lib/tasks/spellcheck_task.rb]: general rule is to use '?' for boolean attributes.

Comment (jreidinger) [inline lib/tasks/spellcheck_task.rb]: from style guide question mark should be used for predicate methods. For me it is just implementation detail if colorize is attribute or somehow magically recognized by term capability.

Comment (lslezak) [inline lib/tasks/spellcheck_task.rb]: OK, I'll switch to `colorize?`, thanks for the feedback.

**Comment by jreidinger** [comment]
LGTM

**Pull request merged**